data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java

```diff
@@ -450,6 +453,51 @@ public void getStateFinalityCheckpoints_shouldGetCheckpointsAfterFinalized()
     verify(mockCombinedChainDataClient).getBestState();
   }
 
+  @Test
+  public void getStateSyncCommittees_shouldGetCommittees()
+      throws ExecutionException, InterruptedException {
+    final Spec altair = TestSpecFactory.createMinimalAltair();
+    final DataStructureUtil dataStructureUtil = new DataStructureUtil(altair);
+    final ChainDataProvider provider =
+        new ChainDataProvider(altair, recentChainData, mockCombinedChainDataClient);
+
+    final SszList<tech.pegasys.teku.spec.datastructures.state.Validator> validators =
+        dataStructureUtil.randomSszList(
+            dataStructureUtil.getBeaconStateSchema().getValidatorsSchema(),
+            16,
+            dataStructureUtil::randomValidator);
+    final SyncCommittee currentSyncCommittee = dataStructureUtil.randomSyncCommittee(validators);
+    final List<UInt64> committeeIndices =
+        List.of(UInt64.valueOf(6), UInt64.valueOf(9), UInt64.valueOf(0));
+
+    final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState internalState =
+        dataStructureUtil
+            .stateBuilderAltair()
+            .validators(validators)
+            .currentSyncCommittee(currentSyncCommittee)
+            .build();
+    when(mockCombinedChainDataClient.getBestState()).thenReturn(Optional.of(internalState));
+
+    final SafeFuture<Optional<StateSyncCommittees>> future =
+        provider.getStateSyncCommittees("head", Optional.empty());
+    assertThat(future).isCompleted();
+    assertThat(future.get().get())
+        .isEqualTo(new StateSyncCommittees(committeeIndices, List.of(committeeIndices)));
+  }
+
+  @Test
+  public void getStateSyncCommittees_shouldRejectPreAltairState() {
+    final ChainDataProvider provider =
+        new ChainDataProvider(spec, recentChainData, combinedChainDataClient);
+    final tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState internalState =
+        data.randomBeaconState();
+    when(mockCombinedChainDataClient.getBestState()).thenReturn(Optional.of(internalState));
+
+    final SafeFuture<Optional<StateSyncCommittees>> future =
+        provider.getStateSyncCommittees("head", Optional.empty());
+    assertThat(future).isCompletedExceptionally();
```

Comment:
Maybe should assert at least the exception type here with `assertThatSafeFuture(future).isCompletedExceptionallyWith`